### PR TITLE
Add decoding of Optimism DAI bridging

### PR DESCRIPTION
**Pull request by nicholasyoder**
DAI uses a different bridge than other tokens.
See https://github.com/makerdao/optimism-dai-bridge

Solves https://github.com/orgs/rotki/projects/11?pane=issue&itemId=68394660

### CODE REVIEW

File: rotkehlchen/chain/ethereum/modules/optimism_bridge/decoder.py [hunk 1]
```diff
@@ -24,7 +24,11 @@
     hex_or_bytes_to_str,
 )
 
-BRIDGE_ADDRESS = string_to_evm_address('0x99C9fc46f92E8a1c0deC1b1747d010903E884bE1')
+BRIDGE_ADDRESSES = (
+    string_to_evm_address('0x99C9fc46f92E8a1c0deC1b1747d010903E884bE1'),  # Normal Bridge
+    string_to_evm_address('0x10E6593CDda8c58a1d0f14C5164B376352a55f2F'),  # L1DAITokenBridge
+)
+OPTIMISM_L1_ESKROW = string_to_evm_address('0x467194771dAe2967Aef3ECbEDD3Bf9a310C76C65')
 OPTIMISM_PORTAL_ADDRESS = string_to_evm_address('0xbEb5Fc579115071764c7423A4f12eDde41f106Ed')
```
Comment (yabirgb) [inline rotkehlchen/chain/ethereum/modules/optimism_bridge/decoder.py:32]: make those constant final

File: rotkehlchen/chain/ethereum/modules/optimism_bridge/decoder.py [hunk 2]
```diff
@@ -121,10 +125,8 @@ def _decode_prove_withdrawal(self, context: DecoderContext) -> DecodingOutput:
     # -- DecoderInterface methods
 
     def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
-        return {
-            BRIDGE_ADDRESS: (self._decode_bridge,),
-            OPTIMISM_PORTAL_ADDRESS: (self._decode_prove_withdrawal,),
-        }
+        return dict.fromkeys(BRIDGE_ADDRESSES, (self._decode_bridge,)) | \
+            {OPTIMISM_PORTAL_ADDRESS: (self._decode_prove_withdrawal,)}
```
Comment (yabirgb) [inline rotkehlchen/chain/ethereum/modules/optimism_bridge/decoder.py:129]: we don't use this syntax. Better use a noqa: E501 hre

File: rotkehlchen/chain/optimism/modules/optimism_bridge/decoder.py [hunk 3]
```diff
@@ -21,7 +21,10 @@
 from rotkehlchen.types import ChainID, ChecksumEvmAddress, EvmTokenKind
 from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
 
-BRIDGE_ADDRESS = string_to_evm_address('0x4200000000000000000000000000000000000010')
+BRIDGE_ADDRESSES = (
+    string_to_evm_address('0x4200000000000000000000000000000000000010'),  # Normal Bridge
+    string_to_evm_address('0x467194771dAe2967Aef3ECbEDD3Bf9a310C76C65'),  # L2DAITokenBridge
+)
```
Comment (yabirgb) [inline rotkehlchen/chain/optimism/modules/optimism_bridge/decoder.py:27]: make them final

File: rotkehlchen/tests/unit/decoders/test_optimism_bridge.py [hunk 4]
```diff
@@ -405,3 +408,155 @@ def test_prove_withdrawal(database, ethereum_inquirer, ethereum_accounts):
...
+    user_address, timestamp, gas_amount, deposit_amount = ethereum_accounts[0], TimestampMS(1723807967000), '0.0006873796276356', '97.27'  # noqa: E501
+    assert events == [
+        EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=Asset('ETH'),
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/decoders/test_optimism_bridge.py:432]:
```suggestion
            asset=A_ETH,
```

File: rotkehlchen/tests/unit/decoders/test_optimism_bridge.py [hunk 5]
```diff
@@ -405,3 +408,155 @@ def test_prove_withdrawal(database, ethereum_inquirer, ethereum_accounts):
...
+            notes=f'Burned {gas_amount} ETH for gas',
+            counterparty=CPT_GAS,
+        ), EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=876,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.WITHDRAWAL,
+            event_subtype=HistoryEventSubType.BRIDGE,
+            asset=Asset('eip155:1/erc20:0x6B175474E89094C44Da98b954EedeAC495271d0F'),
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/decoders/test_optimism_bridge.py:526]:
```suggestion
            asset=A_DAI,
```

File: rotkehlchen/tests/unit/decoders/test_optimism_bridge.py [hunk 6]
```diff
@@ -405,3 +408,155 @@ def test_prove_withdrawal(database, ethereum_inquirer, ethereum_accounts):
...
+            counterparty=CPT_OPTIMISM,
+            address=OPTIMISM_L1_ESKROW,
+        ),
+    ]
+
+
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('optimism_accounts', [['0x301605C95acbED7A1fD9C2c0DeEe964e2AFBd0C3']])
+def test_withdraw_dai_on_optimism(database, optimism_inquirer, optimism_accounts):
+    """DAI uses special bridge. See https://github.com/makerdao/optimism-dai-bridge"""
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/decoders/test_optimism_bridge.py:539]: no need to copy paste the same docstring in all the tests. I would add it to the docstring of `_decode_bridge` where it is more useful and remove it from the tests

File: rotkehlchen/tests/unit/decoders/test_optimism_bridge.py [hunk 7]
```diff
@@ -405,3 +408,151 @@ def test_prove_withdrawal(database, ethereum_inquirer, ethereum_accounts):
...
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('ethereum_accounts', [['0x9531C059098e3d194fF87FebB587aB07B30B1306']])
+def test_withdraw_dai_on_ethereum(database, ethereum_inquirer, ethereum_accounts):
+    evmhash = deserialize_evm_tx_hash('0x3e958996b7c4b2466ca9521d4c688ff4817c28d66bb3f52ec1174c6d0f1c44db')  # noqa: E501
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=ethereum_inquirer,
+        database=database,
+        tx_hash=evmhash,
+    )
+    user_address, timestamp, gas_amount, recieve_amount = ethereum_accounts[0], TimestampMS(1671053711000), '0.0097257501', '106317.949168317256453804'  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/test_optimism_bridge.py:502]:
```suggestion
    user_address, timestamp, gas_amount, receive_amount = ethereum_accounts[0], TimestampMS(1671053711000), '0.0097257501', '106317.949168317256453804'  # noqa: E501
```

**Review by LefterisJP — APPROVED** [review]
LGTM!

**Pull request merged**